### ISSUE TITLE
CosmosDB Target Property (domain) is different to actual URL

### ISSUE THREAD

**Issue by NoelAbrahams** [microsoft/ApplicationInsights-node.js#675]
**AI Version**: 1.8.5
**Node Version**: 13.11.0

### Description of the Problem
We are trying to exclude the automatic collection of Azure CosmosDB dependency tracking via this code

```ts

type EnvelopeType = 'EventData' | 'RequestData' | 'RemoteDependencyData';

ai.defaultClient.addTelemetryProcessor((envelope, _context) => {

    if ((envelope?.data?.baseType as EnvelopeType) === 'RemoteDependencyData') {

        const base = envelope.data as ai.Contracts.Data<Partial<ai.Contracts.RemoteDependencyData>>;
        const request = base.baseData;

        if (request) {

            // Filter out CosmosDB since it has a custom handler 
            if (request.target?.includes('my-cosmos-db.documents.azure.com:443')) {
                return false;
            }
           }
        }
    }

    return true;
});

```

### Expected Behaviour
The handler above should ensure that all calls to `my-cosmos-db.documents.azure.com:443` are excluded from AI logs.

### Actual Behaviour
This is failing because `request.target` is not the same URL as the one to which the calls are being made.

The `request.target` is something like this `my-cosmos-db-datacentre.documents.azure.com:443`

In short, your code (or CosmosDB code) seems to be adding a strange 'datacentre' segment to the URL (Note: the actual data-centre name is used).

Are you able to shed some light on this?

I'm aware that I could try `.documents.azure.com:443`. But ideally there should be an easy way to exclude specific well-known dependencies like CosmosDB. At present, as seen by the code above, any changes to the URL or to the `target` property will break the code.

Thanks

**Comment by markwolff** [comment]
This app insights SDK doesn't modify the URLs it sees at all, and so the cosmosDB driver is probably inserting this region info before it makes the underlying HTTP call.

I think your proposal of custom searching for `*.documents.azure.com:443` is the direction I would go here.

**Comment by NoelAbrahams** [comment]
@markwolff,  

The code we are forced to write here is hacky and as noted here https://github.com/microsoft/ApplicationInsights-node.js/issues/606 the typing is also not helpful. 

There should be an easier way to selectively exclude logging from specific services — especially seeing they are Azure services.

**Comment by hectorhdzg** [comment]
@NoelAbrahams let us know if you are still having issues excluding the data, telemetry processors are the supported way to handle this kind of scenarios, will take a look at typing issue.

**Comment by NoelAbrahams** [comment]
@hectorhdzg 

Since the issue is not fixed, strangely I'm still having this problem, as most people with cognitive ability would expect.

What's your reason for closing this issue? Are you going to 'take a look at typing issue' in your own private repository?

**Comment by hectorhdzg** [comment]
@NoelAbrahams I closed this issue because the main problem is not related to this SDK, like Mark mentioned before we don't update the URLs, if your ask is about better integration with Azure SDKs it would be good to track as a different issue with more details, there is another recent issue regarding allowing filtering for specific endpoints that could also apply here

#712 

Please feel free to reopen and add more details if you believe this is not resolved.

**Comment by NoelAbrahams** [comment]
@hectorhdzg 

> Please feel free to reopen and add more details if you believe this is not resolved.

Perhaps you're not aware: non-collaborators do not have the privilege to reopen an issue closed by a collaborator. So, closing an issue without providing adequate information comes across as authoritarian and rude.


However, thank you for the additional information. I will follow up when I come round to this.
